### PR TITLE
enable sub_array_handler to take a Dimension object as indexes, which…

### PR DESCRIPTION
**Pull request by JakobBD**
… eases working with sub-dimensions such as historic time. 

For application syntax, see [the `test_dimension_subsets()` function](https://github.com/JakobBD/sodym/blob/9cd4da97d3e22a3a18a769a1b5caec026c09d91c/tests/test_named_dim_arrays.py#L144-L166).

### CODE REVIEW

File: sodym/dimensions.py [hunk 1]
```diff
@@ -117,6 +115,28 @@ def expand_by(self, added_dims: list[Dimension]) -> 'DimensionSet':
             raise ValueError('DimensionSet already contains one or more of the dimensions to be added.')
         return DimensionSet(dim_list=self.dim_list + added_dims)
 
+    def drop(self, key: str, inplace: bool=False):
+        dim_to_drop = self._dict[key]
+        if inplace:
+            self.dim_list.remove(dim_to_drop)
+            return
+        else:
```
Comment (SallyDa) [inline sodym/dimensions.py:123]: It is just a stylist thing, but honestly I like my formulation better, with no empty return statement and no unnecessary else clause. Since this is just a small function, it doesn't really matter: readability is fine in both cases. But if there was more stuff in the else (i.e. we have exception cases at the top with return/raise things, and the bulk of the function below), including other clauses with indentations, then I would definitely prefer to leave out the `else` to improve readability by reducing the level of indentations.

**Review by SallyDa — APPROVED** [review]
Looks like a nice useful addition :)

File: sodym/dimensions.py (hunk 1 above)
Comment (JakobBD) [reply sodym/dimensions.py:123]: I asked ChatGPT "I have two code snippets. Please tell me which one is better python style and why:" and got a way too long answer: 

Snippet 2 is generally better Python style. Here’s why:

### 1. **Avoiding Unnecessary Negation**
   - In Snippet 2, the logic is more readable because the `inplace` flag is checked directly with `if inplace:` rather than using `if not inplace:` in Snippet 1.
   - Avoiding negations (like `if not`) in conditional checks typically enhances readability, especially for more complex conditions.

### 2. **Consistent `return` Placement**
   - Snippet 2 clearly separates the cases with an `else` block, making it obvious that only one path will be taken.
   - Although Python allows not using `else` after a return, including it here makes it explicit that the `else` condition is an alternative path. This is slightly more readable and makes it easier to modify or extend the code.

### 3. **Readability and Intent**
   - Snippet 2 reads in a more logical, step-by-step way. The `inplace` flag is checked first, which immediately indicates whether the object will be modified directly or a new instance will be returned.

In summary, Snippet 2 improves clarity by avoiding negation, using a direct conditional structure, and explicitly separating code paths. This makes the intent of the function easier to understand and more maintainable.

Comment (SallyDa) [reply sodym/dimensions.py:123]: There are arguments both ways, and as I said above, in this case it doesn't really matter.

Comment (JakobBD) [reply sodym/dimensions.py:123]: I find point 3 the most important. The if-else structure is clearer that way. It's not the "if not inplace, don't do a lot and return" but really two different paths depending on the condition. I agree that lots of indentations are bad, but the two additional indented lines seem worth it here. If it was more, I'd put it in its own function to decrease indentation.

Comment (JakobBD) [reply sodym/dimensions.py:123]: Sorry for the harshness. I actually do see the advantages of your version as well, and also like that style in some cases.
